homeassistant/components/konnected.py

```diff
@@ -0,0 +1,293 @@
+"""
+Support for Konnected devices.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/konnected/
+"""
+import asyncio
+import logging
+import voluptuous as vol
+
+from aiohttp.hdrs import AUTHORIZATION
+from aiohttp.web import Request, Response  # NOQA
+
+from homeassistant.components.binary_sensor import DEVICE_CLASSES_SCHEMA
+from homeassistant.components.discovery import SERVICE_KONNECTED
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.const import (
+    HTTP_BAD_REQUEST, HTTP_INTERNAL_SERVER_ERROR, HTTP_UNAUTHORIZED)
+from homeassistant.helpers import discovery
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['konnected==0.1.2']
+
+DOMAIN = 'konnected'
+
+PIN_TO_ZONE = {1: 1, 2: 2, 5: 3, 6: 4, 7: 5, 8: 'out', 9: 6}
+ZONE_TO_PIN = {zone: pin for pin, zone in PIN_TO_ZONE.items()}
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: {
+            'auth_token': str,
+            vol.Optional('home_assistant_url'): str,
+            'devices': [{
+                vol.Required('id', default=''): vol.Coerce(str),
+                'sensors': [{
+                    vol.Exclusive('pin', 's_pin'): vol.Any(*PIN_TO_ZONE),
+                    vol.Exclusive('zone', 's_pin'): vol.Any(*ZONE_TO_PIN),
+                    vol.Required('type', default='motion'):
+                        DEVICE_CLASSES_SCHEMA,
+                    vol.Optional('name'): str,
+                }],
+                'actuators': [{
+                    vol.Exclusive('pin', 'a_pin'): vol.Any(*PIN_TO_ZONE),
+                    vol.Exclusive('zone', 'a_pin'): vol.Any(*ZONE_TO_PIN),
+                    vol.Optional('name'): str,
+                    vol.Required('activation', default='high'):
+                        vol.All(vol.Lower, vol.Any('high', 'low'))
+                }],
+            }],
+        },
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+DEPENDENCIES = ['http']
+
+ENDPOINT_ROOT = '/api/konnected'
+UPDATE_ENDPOINT = (
+    ENDPOINT_ROOT +
+    r'/device/{device_id:[a-zA-Z0-9]+}/{pin_num:[0-9]}/{state:[01]}')
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Set up the Konnected platform."""
+    cfg = config.get(DOMAIN)
+    if cfg is None:
+        cfg = {}
+
+    auth_token = cfg.get('auth_token') or 'supersecret'
```

Comment:
Is 'supersecret' the default? If yes, then pass it in the validation.

Reply:
I opted to remove the default and make it required for the user to set.